src/Eagle.ts

```diff
@@ -3241,12 +3256,34 @@ export class Eagle {
          $("#"+divID).hide();
     }
 
-    addEmptyField = (node:Node) : void => {
-        node.addEmptyField(this.selectedNode())
-    }
+    addEmptyTableRow = () : void => {
+        var fieldIndex
+        if(Eagle.parameterTableSelectionParentId() != -1){
+        // A cell in the table is selected well insert new row instead of adding at the end
+            fieldIndex = Eagle.parameterTableSelectionParentId()+1
+            if(Eagle.parameterTableType() === 'component'){
```

Comment:
Maybe Eagle.parameterTableType could be of type Eagle.FieldType, instead of string.

If you want to reset Eagle.parameterTableType to neither 'component' or 'application', you would normally set it to ''. But if you make the change to use a Eagle.FieldType instead, you may want to add a Eagle.FieldType.Unknown value to the enum.